o1vm/src/cli/cannon.rs

```diff
@@ -104,9 +99,25 @@ pub struct RunArgs {
     pub vm_cfg: MipsVmConfigurationArgs,
 }
 
+#[derive(Parser, Debug, Clone)]
+pub struct GenStateJsonArgs {
+    #[arg(short = 'i', long, value_name = "FILE", help = "input ELF file")]
+    pub input: String,
+    #[arg(
+        short = 'o',
+        long,
+        value_name = "FILE",
+        default_value = "state.json",
+        help = "output state.json file"
+    )]
+    pub output: String,
+}
+
 #[derive(Subcommand, Clone, Debug)]
 pub enum Cannon {
     Run(RunArgs),
     #[command(name = "test-optimism-preimage-read")]
     TestPreimageRead(RunArgs),
+    #[command(name = "gen-state-json")]
+    GenStateJson(GenStateJsonArgs),
```

Comment:
I'm fine with an additional subcommand to test the generation of `state.json`. However, this will only be useful for Cannon programs, not for generic programs. `state.json` won't be used for other MIPS/RISC-V programs.
MIPS/RISC-V program state should be loaded directly from the ELF, see [this example](https://github.com/o1-labs/proof-systems/blob/master/o1vm/tests/test_riscv_elf.rs#L113).

Reply:
:ok_hand: -- we can remove it in the future as we want, but right now I am using it heavily for local development/testing